python_modules/dagster/dagster/_core/storage/db_io_manager.py

```diff
@@ -232,7 +232,9 @@ def _get_table_slice(
                     partition_dimensions.append(
                         TablePartitionDimension(
                             partition_expr=cast(str, partition_expr),
-                            partitions=context.asset_partitions_time_window,
+                            partitions=context.asset_partitions_time_window
```

Comment:
Fixes a small bug that I encountered.

Previously, the time window partition mapping would return time windows with no partitions (the start == the end). This PR fixes this behavior, but then `context_asset_partitions_time_window` would throw an error in this case. This check passes in an empty list of partitions if no partitions exist.